Sources/ServiceLifecycle/Service.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftServiceLifecycle open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftServiceLifecycle project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftServiceLifecycle project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// This is the basic protocol that a service has to implement.
+public protocol Service: Sendable {
+    /// This indicates if the service is expected to be running for the entire duration.
+    /// If a long running service is returning from the ``Service/run()-1ougx`` method it is treated
+    /// as an unexpected early exit and all other services are going be cancelled.
+    ///
+    /// By default this is set to `true`.
+    var isLongRunning: Bool { get }
```

Comment:
I am not the biggest fan of this one. Especially how it interacts with the `run` method. I was thinking that we could introduce two protocols. One for long running services and the other for short running services. Open to suggestions here!

Reply:
_if_ we need to know this to drive behavior, then a protocol seems more flexible and future facing

Reply:
Not sure I'm a massive fan of the name of this.  Also wonder if there is a need for it.